### PR TITLE
Create <Heading /> Component

### LINKED ISSUE #3: Create <Heading />
[Design System Link](https://www.notion.so/h4iuiuc/Hack4Impact-s-Design-System-Bridge-40aaae97c71f400084b4bbcc5cd8cdee#92c6a679980b4ea3afb4431fe80720d9)

Heading component that matches h1 styling

**props**
```
- type
    - h1, h2, h3, h4
```

Should also respect `as`

### PR DESCRIPTION
**Pull request by jackieo5023**
This PR:
* Creates the `<Heading />` Component
    * `as` is **not** listed as a PropType
* Moves "subtitle" to be under `<Text />`

I wasn't sure as to the best way to go about coding the conditional render of `<Heading />` - very open to suggestions.

Resolves #3

### CODE REVIEW

**Comment by tko22** [comment]
hm what if you did something like this

```js
const HeadingBase = styled.h2`
 // styles here
`

const Heading = ({type, ...props}) → {
     if (type == "h1") {
            <Button as="h1" type={type} {...props} >
     }
     // ... etc
}
```
so dev doesnt know the `as` prop, we just make use of it internally

File: src/components/Heading/Heading.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+import { variant } from 'styled-system';
+import PropTypes from 'prop-types';
+import theme from '../../theme';
+import { get } from '../../utils/utils';
+import { TYPOGRAPHY, COMMON } from '../../utils/constants';
+
+const style = css`color: ${get('colors.text.coolTone.main')};
```
Comment (tko22) [inline src/components/Heading/Heading.js:9]: woah cool

File: src/theme.js [hunk 2]
```diff
@@ -232,6 +225,13 @@ const typography = {
     letterSpacing: letterSpacings.half, // 0.5px
     lineHeight: '40px',
   },
+  subtitle: {
```
Comment (tko22) [inline src/theme.js:228]: thx :)

File: playground/Playground.js [hunk 3]
```diff
@@ -2,7 +2,9 @@
 import React from 'react';
 import { colors } from '../src/theme';
 
-import { Button, Link, Text } from '../src';
+import {
+  Button, Link, Text, Heading,
+} from '../src';
```
Comment (tko22) [inline playground/Playground.js:7]: adding this is fine to me tho lol

File: playground/Playground.js [hunk 4]
```diff
@@ -15,7 +17,8 @@ export function Playground() {
       <div>
         {/* Add the code you want to test here */}
         <Button href="https://google.com">APPLY</Button>
-        <Link variant="success" href="https://google.com">New link</Link>
+        <Link variant="primary" href="https://google.com">New link</Link>
```
Comment (tko22) [inline playground/Playground.js:20]: maybe dont commit the playground? I shouldnt of committed before too.. I'll remove the other stuff later

File: src/components/Heading/Heading.js (hunk 1 above)
Comment (jackieo5023) [reply src/components/Heading/Heading.js:9]: i deleted it but i guess it's useful for later :')

**Pull request merged**